async_fuse/src/util.rs

```diff
@@ -72,3 +77,73 @@ pub fn mode_from_kind_and_perm(kind: SFlag, perm: u16) -> u32 {
         file_perm | ftype
     }
 }
+
+/// polyfill from smol 0.1.18 to 1.2.5
+#[allow(dead_code)]
+pub async fn unblock<T, F>(f: F) -> T
+where
+    F: FnOnce() -> T + Send + 'static,
+    T: Send + 'static,
+{
+    smol::Task::blocking(async move { f() }).await
+}
+
+/// Stores short bytes on stack, stores long bytes on heap and provides [`CStr`]
```

Comment:
What's the benefit of storing bytes on stack? It requires another copy from the code.

Reply:
Copy is unavoidable when passing bytes to a ffi function which needs c-str, unless the bytes are nul-terminated.
We can reduce allocation by storing short bytes on stack like what `with_nix_path` does.
<https://docs.rs/nix/0.19.1/src/nix/lib.rs.html#234-253>
`with_c_str` returns `io::Result<T>` and will not reject bytes whose length is greater than `PATH_MAX`

Reply:
Pleae tell the benefit of stack allocation. And comment the stack buf size.